Formula/nginx-unit.rb

```diff
@@ -0,0 +1,223 @@
+    # prefix-relative directory for loading and install moduless to keg-local
+    # to be linked in so other formulae can also link modules.
+    system "./configure", "--prefix=#{prefix}",
+                          "--modules=#{HOMEBREW_PREFIX}/lib/unit",
+                          "--state=#{var}/unit",
+                          "--log=#{var}/log/unit.log",
+                          "--pid=#{var}/run/unit.pid",
+                          "--control=unix:#{var}/run/unit.control.sock",
+                          "--cc-opt=-I#{pcre.opt_include} -I#{openssl.opt_include}",
+                          "--ld-opt=-L#{pcre.opt_lib} -L#{openssl.opt_lib}"
+
+    system "make", "install"
+
+    libunit = lib/"unit"
+    libunit.mkpath
+
+    system "./configure", "perl"
+    system "make", "perl"
+    libunit.install "build/perl.unit.so"
+
+    system "./configure", "python"
+    system "make", "python"
+    libunit.install "build/python.unit.so"
+
+    system "./configure", "ruby"
+    system "make", "ruby"
+    libunit.install "build/ruby.unit.so"
+  end
+
+  def post_install
+    (var/"run").mkpath
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using
+      curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+
+      Modules are available for system perl, python and ruby. The system php doesn't
+      support nginx-unit. You can install additional modules into:
+        #{HOMEBREW_PREFIX/"lib/unit"}
+    EOS
+  end
+
+  plist_options :manual => "#{HOMEBREW_PREFIX}/sbin/unitd --no-daemon"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{opt_sbin}/unitd</string>
+            <string>--no-daemon</string>
+        </array>
+        <key>WorkingDirectory</key>
+        <string>#{HOMEBREW_PREFIX}</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    require "json"
+    require "net/http"
+    require "socket"
+    require "timeout"
+
+    begin
+      # We need to install rack for the ruby module test to work, so set a
+      # temporary gems path before booting unitd. (unitd boots and functions
+      # fine without, but fails to start an app until rack is installed.)
+      # And we install old rack for el capitan / sierra compat.
+      rubygems = testpath/"ruby/gems"
+      system "gem", "install", "rack", "-v", "1.6.10", "--install-dir=#{rubygems}"
+      ENV["GEM_PATH"] = rubygems
+
+      control = testpath/"unit.control.sock"
+
+      pid = spawn "#{sbin}/unitd", "--no-daemon", "--log", "/dev/stdout", "--control", "unix:#{control}"
```

Comment:
wrap to avoid going beyond the 80th column

Reply:
Done!